desc/internal/util.go

```diff
@@ -219,17 +219,17 @@ const (
 )
 
 // JsonName returns the default JSON name for a field with the given name.
+// This mirrors the algorithm in protoc:
+//  https://github.com/protocolbuffers/protobuf/blob/v21.3/src/google/protobuf/descriptor.cc#L95
 func JsonName(name string) string {
 	var js []rune
 	nextUpper := false
-	for i, r := range name {
+	for _, r := range name {
 		if r == '_' {
 			nextUpper = true
 			continue
 		}
-		if i == 0 {
-			js = append(js, r)
```

Comment:
Whoops. Looks like I may have accidentally been looking at `ToCamelCase` where `lower_first` param was true, instead of at `ToJsonName` in the protoc codebase ([reference](https://github.com/protocolbuffers/protobuf/blob/v21.3/src/google/protobuf/descriptor.cc#L95)).

Luckily, this bit was basically a no-op: it appears to be trying to skipping the capitalization of the first rune, but if the first rune of the result were to be capitalized, it means that the first rune in `name` is actually `_`, so this was basically a no-op since `nextUpper` could never be true when `i == 0`...

I've added some fields to a test proto (which gets compared to `protoc` output) to make sure that we are handling weird cases correctly, such as field that starts with `_` and... a field whose whole name is `_` (in which case the `json_name` attribute is the empty string! 😨).

Reply:
Wow.. great find. I am just so sad that we have to maintain all of this nonsense.